dashboard/app/controllers/maker_controller.rb

```diff
@@ -27,4 +27,25 @@ def apply
 
     render json: {eligible: application.eligible_unit_6_intention?}
   end
+
+  def schoolchoice
+    school_id = params.require(:nces)
+
+    school = School.find(school_id)
+
+    # Must have started an application, and have said they were teaching unit 6
+    application = CircuitPlaygroundDiscountApplication.find_by_studio_person_id(current_user.studio_person_id)
+    return head :not_found unless application
+    return head :forbidden unless application.eligible_unit_6_intention?
+
+    # Update user's school if provided school_id is different
+    current_school_id = current_user.try(:school_info).try(:school_id)
+    if school_id != current_school_id
+      current_user.update!(school_info: SchoolInfo.find_by_school_id(school_id))
+    end
+
+    application.update!(has_confirmed_school: true, full_discount: school.high_needs?)
```

Comment:
It looks like our client is designed to only allow the user to set their school once and then lock them out of changing it, but this route would accept new school information as many times as you sent it.  I'd expect this route to fail (maybe 400 BAD_REQUEST?) if you tried to change an application school, unless the change is made by an admin.

Reply:
updated line 39 to also return forbidden if we'd already confirmed school